x/xvalidator/validator.go

```diff
@@ -0,0 +1,29 @@
+package xvalidator
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/mesg-foundation/core/x/xnet"
+	validator "gopkg.in/go-playground/validator.v9"
+)
+
+// IsDomainName validates if given field is valid domain name.
+func IsDomainName(fl validator.FieldLevel) bool {
+	if fl.Field().String() == "" {
+		return true
+	}
+	return xnet.IsDomainName(fl.Field().String())
+}
+
+// IsPort validates if given field is valid port.
+func IsPort(fl validator.FieldLevel) bool {
+	ports := strings.Split(fl.Field().String(), ":")
+	for _, port := range ports {
+		i, err := strconv.Atoi(port)
+		if !(err == nil && 0 < i && i <= 65535) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
IsPort is now compatible with port notation like xx:yy as the service.start function is also compatible with

Reply:
It has a bug, you can write port in format `80:80:80` and it will be ok.

Also it should be renamed to `IsPortMapping` and register under `port_map(ing?)` tag

Reply:
Also I used value like 65535 and it will be good to change them to constant.

We can think of adding `/tcp` `/udp` to validation but it's not required in this PR

Reply:
Done. Let's do the `/tcp` and `/udp` in another PR with also the implementation in `service.Start`